fdb-extensions/src/main/java/com/apple/foundationdb/async/hnsw/HNSW.java

```diff
@@ -0,0 +1,1945 @@
+     * @param executor the non-null {@link Executor} for concurrent operations, such as building the graph.
+     * @param numDimensions the number of dimensions
+     */
+    public HNSW(@Nonnull final Subspace subspace, @Nonnull final Executor executor, final int numDimensions) {
+        this(subspace, executor, DEFAULT_CONFIG_BUILDER.build(numDimensions), OnWriteListener.NOOP, OnReadListener.NOOP);
+    }
+
+    /**
+     * Constructs a new HNSW graph instance.
+     * <p>
+     * This constructor initializes the HNSW graph with the necessary components for storage,
+     * execution, configuration, and event handling. All parameters are mandatory and must not be null.
+     *
+     * @param subspace the {@link Subspace} where the graph data is stored.
+     * @param executor the {@link Executor} service to use for concurrent operations.
+     * @param config the {@link Config} object containing HNSW algorithm parameters.
+     * @param onWriteListener a listener to be notified of write events on the graph.
+     * @param onReadListener a listener to be notified of read events on the graph.
+     *
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     */
+    public HNSW(@Nonnull final Subspace subspace,
+                @Nonnull final Executor executor,
+                @Nonnull final Config config,
+                @Nonnull final OnWriteListener onWriteListener,
+                @Nonnull final OnReadListener onReadListener) {
+        this.random = new Random(config.getRandomSeed());
+        this.subspace = subspace;
+        this.executor = executor;
+        this.config = config;
+        this.onWriteListener = onWriteListener;
+        this.onReadListener = onReadListener;
+    }
+
+
+    /**
+     * Gets the subspace associated with this object.
+     *
+     * @return the non-null subspace
+     */
+    @Nonnull
+    public Subspace getSubspace() {
+        return subspace;
+    }
+
+    /**
+     * Get the executor used by this hnsw.
+     * @return executor used when running asynchronous tasks
+     */
+    @Nonnull
+    public Executor getExecutor() {
+        return executor;
+    }
+
+    /**
+     * Get this hnsw's configuration.
+     * @return hnsw configuration
+     */
+    @Nonnull
+    public Config getConfig() {
+        return config;
+    }
+
+    /**
+     * Get the on-write listener.
+     * @return the on-write listener
+     */
+    @Nonnull
+    public OnWriteListener getOnWriteListener() {
+        return onWriteListener;
+    }
+
+    /**
+     * Get the on-read listener.
+     * @return the on-read listener
+     */
+    @Nonnull
+    public OnReadListener getOnReadListener() {
+        return onReadListener;
+    }
+
+    @Nonnull
+    @SuppressWarnings("PMD.UseUnderscoresInNumericLiterals")
+    RealVector centroidRot(@Nonnull final FhtKacRotator rotator) {
+        final double[] centroidData = {29.0548, 16.785500000000003, 10.708300000000001, 9.7645, 11.3086, 13.3,
```

Comment:
What is this doing here? Is this some testing artifact that needs to be removed?